Tools/automation.ps1

```diff
@@ -0,0 +1,790 @@
+    }
+    Process
+    {
+        ## Determines if the Resource Group already exists
+        $Result = Get-AzResourceGroup -Name $AzResourceGroupName -ErrorAction SilentlyContinue -ErrorVariable RGerror -InformationAction SilentlyContinue -WarningAction SilentlyContinue
+
+        IF($RGerror)
+        { 
+            ## Resource Group does not already exist, creating a new Resource Group
+            Write-Host "Creating Resource Group ($AzResourceGroupName) in the $AzLocation region..."
+            $Result = New-AzResourceGroup -Name $AzResourceGroupName -Location $AzLocation 
+        }
+    }
+    End
+    {
+        IF(!$RGerror)
+        {
+            ## Resource Group already exists, do nothing
+            Write-Host "Resource Group $AzResourceGroupName already exists, will not recreate..." -ForegroundColor Yellow
+            Return $true
+        }
+        ElseIF(Get-AzResourceGroup -Name $AzResourceGroupName -ErrorAction SilentlyContinue -WarningAction SilentlyContinue -InformationAction SilentlyContinue)
+        {
+            ## Resource Group was created Successfully 
+            Write-Host "  Resource Group ($AzResourceGroupName) in the $AzLocation region was created successfully"
+            Return $true
+        }
+        else 
+        {
+            ## Resource Group failed to be created 
+            Write-Host "  Resource Group ($AzResourceGroupName) in the $AzLocation region failed to be created" -ForegroundColor Red 
+            Return $false
+        }
+    }
+}
+
+Function New-ARMParameterObject
+{
+    param(
+        [Parameter(Position=0, Mandatory=$true)] [string]$ParameterFolderPath,
+        [Parameter(Position=1, Mandatory=$true)] [string]$TemplateFolderPath,
+        [Parameter(Position=2, Mandatory=$false)][string]$Index,
+        [Parameter(Position=3, Mandatory=$true)] [string]$ResourcePrefix,
+        [Parameter(Position=4, Mandatory=$true)] [string]$AzLocation
+    )
+    Begin
+    {
+        ## The Names that are to be assigned to each resource.
+        $AppInsightsName    = $($ResourcePrefix + "appinsight" + $Index)            # Name that will be assigned to the Azure AppInsights.
+        $KeyVaultName       = $($ResourcePrefix + "keyvault" + $Index)              # Name that will be assigned to the Azure Keyvault.
+        $StorageAccountName = $($ResourcePrefix + "sa" + $Index).Replace("-", "")   # Name that will be assigned to the Azure Storage Account. Doesn't support special characters.
+        $aspName            = $($ResourcePrefix + "asp" + $Index)                   # Name that will be assigned to the Azure ASP.
+        $CDBAccountName     = $($ResourcePrefix + "cdba" + $Index)                  # Name that will be assigned to the Azure Cosmos Database Account.
+        $CDBDatabaseName    = $("WinGet")                                           # Name of the Cosmos Database - Value must match the name found in the Compiled Windows Package Manager Functions (CompiledFunctions.zip).
```

Comment:
I plan to make these 2 configurable as well.

Reply:
After they've become configurable, we can offer to rename or remove the comment stating that these must be the specified values.

Reply:
I recommend taking these out of here for now, and passing them in as variables and then set the WinGet and Manifests defaults closer to the outside of the code.